subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1409,16 +1409,26 @@ uint8_t ll_adv_enable(uint8_t enable)
 				sync_is_started = 1U;
 
 				lll_adv_data_enqueue(lll, pri_idx);
+			} else {
+				/* TODO: Find the anchor before the group of
+				 *       active Periodic Advertising events, so
+				 *       that auxiliary sets are grouped such
+				 *       that auxiliary sets and Periodic
+				 *       Advertising sets are non-overlapping
+				 *       for the same event interval.
+				 */
 			}
 #endif /* CONFIG_BT_CTLR_ADV_PERIODIC */
 
 			/* Keep aux interval equal or higher than primary PDU
 			 * interval.
 			 */
-			aux->interval = adv->interval +
-					(HAL_TICKER_TICKS_TO_US(
-						ULL_ADV_RANDOM_DELAY) /
-						ADV_INT_UNIT_US);
+			aux->interval =
+				ceiling_fraction(((uint64_t)adv->interval *
+						  ADV_INT_UNIT_US) +
+						 HAL_TICKER_TICKS_TO_US(
+							ULL_ADV_RANDOM_DELAY),
+						 CONN_INT_UNIT_US);
```

Comment:
This function is hard to get an overview of and understand (it is a stunning 800 lines of code!), so please enlighten me: Is this change related to connectable ADV, or are you reusing `CONN_INT_UNIT_US` as that is the same unit (1.25 ms)?

Reply:
Add comment.

Reply:
I got to admit that I do not understand the comment that well. 

I understand that using the unit of 1.25ms groups it with ACL and periodic advertising events, because they share the same unit of interval (1.25ms), but I don't understand how a unit can represent a behavior: 

```
Use connection interval units to represent the periodic behavior of scheduling
```

Would it make sense to create a `PER_ADV_INT_UNIT_US` as well? At least the semantic meaning of the unit will be there.